library/std/src/sync/poison.rs

```diff
@@ -267,6 +313,7 @@ where
 {
     match result {
         Ok(t) => Ok(f(t)),
+        #[cfg(panic = "unwind")]
```

Comment:
`map_result` isn't public API so instead of removing the variant you could add an `unreachable_unchecked` in this